### PR TITLE
Upgrade packages

### LINKED ISSUE #32: Inconsistent JVM-target compatibility
I have a problem similar to this Issue(#29).

Whether it occurs or not seems to depend on the kotlin version.
The build seems to succeed if the version of Kotlin is 1.7.X and fail if the version is 1.8.X or higher.

* android/settings.gradle
```android/settings.gradle
plugins {
    ...
//    // -> Build Success
//    id "org.jetbrains.kotlin.android" version "1.7.10" apply false

    // -> Build Failed
    id "org.jetbrains.kotlin.android" version "1.8.22" apply false
}
```

* android/app/build.gradle
```android/app/build.gradle
android {
    ...
    compileOptions {
        sourceCompatibility = JavaVersion.VERSION_17
        targetCompatibility = JavaVersion.VERSION_17
    }

    kotlinOptions {
        jvmTarget = JavaVersion.VERSION_17
    }
}
```

* android/gradle/wrapper/gradle-wrapper.properties
```android/gradle/wrapper/gradle-wrapper.properties
distributionUrl=https\://services.gradle.org/distributions/gradle-8.9-all.zip
```

---
### Build Result
```.bat
Launching lib\main.dart on XXXXXX in debug mode...
Running Gradle task 'assembleDebug'...

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':async_preferences:compileDebugKotlin'.
> 'compileDebugJavaWithJavac' task (current target is 1.8) and 'compileDebugKotlin' task (current target is 17) jvm target compatibility should be set to the same Java version.
  Consider using JVM toolchain: https://kotl.in/gradle/jvm/toolchain


* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 28s
Error: Gradle task assembleDebug failed with exit code 1
```

Many people who have updated to Android Studio Ladybug have encountered the following problem, and will probably take action to increase the Gradle version, and while at it, increase the kotlin version.
https://github.com/flutter/flutter/issues/156307
https://github.com/flutter/flutter/issues/156111
Also, since google_mobile_ads: 5.2.0 requires kotlin version 1.8.X or higher, I think many people will encounter this problem given that many people use this library with respect to advertising.
Therefore, we would appreciate it if you could address this issue.

### PR DESCRIPTION
**Pull request by minhdanh**
This should fix #32

### CODE REVIEW

**Comment by svprdga** [comment]
Thank you for your contribution @minhdanh , before proceeding to the PR review, please modify the target branch of this pull request. You should target the `develop` branch. Once modified I will do a full set of tests and I will proceed to create a new version of the plugin.

Thank you.

**Comment by minhdanh** [comment]
@svprdga Thanks. I've just updated

File: pubspec.yaml
Comment (svprdga) [inline pubspec.yaml:1]: You are forcing a specific Dart SDK with these changes. Given that you are not altering any Dart code, the changes in this file should not be required.

File: pubspec.lock
Comment (svprdga) [inline pubspec.lock:1]: As the changes in pubspec.yaml should not be required, changes in this file should also not be required.

**Review by svprdga — CHANGES_REQUESTED** [review]

File: pubspec.lock
Comment (minhdanh) [reply pubspec.lock:1]: I reverted that change. Thanks!

File: pubspec.yaml
Comment (minhdanh) [reply pubspec.yaml:1]: Reverted

**Review by svprdga — CHANGES_REQUESTED** [review]
It seems that the example project cannot compile now, I guess that the gradle dependencies should be upgraded as well there.

Please try to fix the example project, maybe by running the AS built-in gradle migration tool is enough.

Thanks.

**Comment by minhdanh** [comment]
@svprdga I've just fixed the example. Can you please check?

**Review by svprdga — APPROVED** [review]
Thanks for your contribution, I'll create a new release soon with the changes and I'll let you know in the related issue.

**Pull request merged**